### PR TITLE
extract more engine related properties

### PR DESCRIPTION
**Pull request by mimmi20**
This PR is the first step to remove deprecated properties from testing.

Properties more related to the rendering engine than to the browser are moved to the `engine.json` and removed from the `user-agents/*.json` files.

The `MajorVer` and `MinorVer` properties are removed from the sources and are generated from the `Version` property.

This PR is related to #1433.

### CODE REVIEW

File: src/Browscap/Data/DataCollection.php [hunk 1]
```diff
@@ -224,6 +224,60 @@ public function addSourceFile($src)
...
+        return $this;
+    }
+
+    /**
+     * @param array                        $divisionData
+     * @param string $src
+     *
+     * @throws \UnexpectedValueException
+     */
+    private function validateDivisionDate(array $divisionData, $src)
```
Comment (jaydiablo) [inline src/Browscap/Data/DataCollection.php]: Should this be named `validateDivisionData`?

Can `$src` be type-hinted as a `string`?

File: src/Browscap/Data/PropertyHolder.php [hunk 2]
```diff
@@ -189,6 +189,37 @@ public function isStandardModeProperty($propertyName, WriterInterface $writer =
     }
 
     /**
+     * Determine if the specified property is an property that should
+     * be included in the "full" versions of the files only
+     *
+     * @param  string $propertyName
+     * @return bool
+     */
+    public function isDeprecatedProperty($propertyName)
```
Comment (jaydiablo) [inline src/Browscap/Data/PropertyHolder.php]: Type hints?

File: resources/user-agents/apps/amaya.json [hunk 3]
```diff
@@ -11,48 +11,37 @@
...
-            "MajorVer": "7"
+            "Version": "7.0"
           }
         },
         {
           "match": "amaya/8.*",
           "properties": {
-            "Version": "8.0",
-            "MajorVer": "8",
-            "CssVersion": "2"
```
Comment (jaydiablo) [inline resources/user-agents/apps/amaya.json]: There is no engine defined on this useragent, won't removing the CssVersion property here change how this one parses?  I assume there aren't any tests covering this pattern.

Same for the ones below.

**Comment by mimmi20** [comment]
This branch will be rebased after PR #1579 is merged. After this all errors will be solved.

**Pull request closed without merging**